x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/AggregateMapper.java

```diff
@@ -131,11 +131,16 @@ private Stream<NamedExpression> map(Expression aggregate, boolean grouping) {
     }
 
     private static List<NamedExpression> computeEntryForAgg(Expression aggregate, boolean grouping) {
+        if (aggregate instanceof ToIntermediateState intermediateStateGetter) {
```

Comment:
An interface like this looked interesting to make custom logics. It could be applied to other edge cases we have around AggregateMapper.

_This doesn't remove the reflection-based logic we use here, it's just an easier-to-use override_